framework/src/modules/token/api.ts

```diff
@@ -158,4 +158,57 @@ export class TokenAPI extends BaseAPI {
 		}
 		await userStore.setWithSchema(address, address, userStoreSchema);
 	}
+
+	public async burn(
+		apiContext: APIContext,
+		senderAddress: Buffer,
+		_id: TokenID,
+		amount: bigint,
+	): Promise<void> {
+		const userStore = apiContext.getStore(this.moduleID, STORE_PREFIX_USER);
+		const sender = await userStore.getWithSchema<UserStoreData>(senderAddress, userStoreSchema);
+		if (sender.availableBalance < amount + this._minBalance) {
+			throw new Error(
+				`Sender ${senderAddress.toString(
+					'hex',
+				)} balance ${sender.availableBalance.toString()} is not sufficient for ${(
+					amount + this._minBalance
+				).toString()}`,
+			);
+		}
+		sender.availableBalance -= amount;
+		await userStore.setWithSchema(senderAddress, sender, userStoreSchema);
+	}
+
+	public async mint(
+		apiContext: APIContext,
+		address: Buffer,
+		_id: TokenID,
+		amount: bigint,
+	): Promise<void> {
+		const userStore = apiContext.getStore(this.moduleID, STORE_PREFIX_USER);
+		let recipient: UserStoreData;
+		try {
+			recipient = await userStore.getWithSchema<UserStoreData>(address, userStoreSchema);
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+			recipient = {
+				availableBalance: BigInt(0),
+				lockedBalances: [],
+			};
+		}
+		if (recipient.availableBalance < amount + this._minBalance) {
```

Comment:
Why this check here ?
Do you mean `if (amount < this_minbalance)` ?

Reply:
good catch. it shouldve been `recipient.availableBalance + amount  < + this._minBalance` to check if the  recipient  will have more  than min balance after receiving  the  token